test/TestProjects/CollapseRequestCondition-LowLevel/Generated/RequestConditionCollapseClient.cs

```diff
@@ -164,6 +364,97 @@ internal HttpMessage CreateCollapseGetRequest(RequestConditions requestCondition
             return message;
         }
 
+        internal HttpMessage CreateMissIfNonMatchGetRequest(RequestConditions requestConditions, RequestContext context)
+        {
+            var message = _pipeline.CreateMessage(context);
+            var request = message.Request;
+            request.Method = RequestMethod.Get;
+            var uri = new RawRequestUriBuilder();
+            uri.Reset(_endpoint);
+            uri.AppendPath("/RequestConditionCollapse/missIfNonMatch", false);
+            request.Uri = uri;
+            if (requestConditions != null)
+            {
+                if (requestConditions.IfNoneMatch != null) throw new NotSupportedException($"IfNoneMatch head is not supported here");
+                request.Headers.Add(requestConditions, "R");
+            }
+            message.ResponseClassifier = ResponseClassifier200.Instance;
+            return message;
+        }
+
+        internal HttpMessage CreateMissIfMatchGetRequest(RequestConditions requestConditions, RequestContext context)
+        {
+            var message = _pipeline.CreateMessage(context);
+            var request = message.Request;
+            request.Method = RequestMethod.Get;
+            var uri = new RawRequestUriBuilder();
+            uri.Reset(_endpoint);
+            uri.AppendPath("/RequestConditionCollapse/missIfMatch", false);
+            request.Uri = uri;
+            if (requestConditions != null)
+            {
+                if (requestConditions.IfMatch != null) throw new NotSupportedException($"IfMatch head is not supported here");
+                request.Headers.Add(requestConditions, "R");
+            }
+            message.ResponseClassifier = ResponseClassifier200.Instance;
+            return message;
+        }
+
+        internal HttpMessage CreateMissIfModifiedSinceGetRequest(RequestConditions requestConditions, RequestContext context)
+        {
+            var message = _pipeline.CreateMessage(context);
+            var request = message.Request;
+            request.Method = RequestMethod.Get;
+            var uri = new RawRequestUriBuilder();
+            uri.Reset(_endpoint);
+            uri.AppendPath("/RequestConditionCollapse/missIfModifiedSince", false);
+            request.Uri = uri;
+            if (requestConditions != null)
+            {
+                if (requestConditions.IfModifiedSince != null) throw new NotSupportedException($"IfModifiedSince head is not supported here");
```

Comment:
Parameter validations should happen inside the client call, before other actions are executed.  (For example, we don't want to start a Diagnostic Scope, if we know the method is going to fail with an ArgumentException.)